lib/ansible/modules/cloud/amazon/ec2_vpc_route_table.py

```diff
@@ -565,145 +614,104 @@ def ensure_route_table_present(connection, module):
+    routes = create_route_spec(connection, module, vpc_id)
 
     changed = False
     tags_valid = False
 
     if lookup == 'tag':
         if tags is not None:
             try:
-                route_table = get_route_table_by_tags(connection, vpc_id, tags)
-            except EC2ResponseError as e:
-                module.fail_json(msg="Error finding route table with lookup 'tag': {0}".format(e.message),
-                                 exception=traceback.format_exc())
-            except RuntimeError as e:
-                module.fail_json(msg=e.args[0], exception=traceback.format_exc())
+                route_table = get_route_table_by_tags(connection, module, vpc_id, tags)
+            except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                module.fail_json_aws(e, msg="Error finding route table with lookup 'tag'")
         else:
             route_table = None
     elif lookup == 'id':
         try:
-            route_table = get_route_table_by_id(connection, vpc_id, route_table_id)
-        except EC2ResponseError as e:
-            module.fail_json(msg="Error finding route table with lookup 'id': {0}".format(e.message),
-                             exception=traceback.format_exc())
+            route_table = get_route_table_by_id(connection, module, route_table_id)
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            module.fail_json_aws(e, msg="Error finding route table with lookup 'id'")
 
     # If no route table returned then create new route table
     if route_table is None:
-        try:
-            route_table = connection.create_route_table(vpc_id, module.check_mode)
-            changed = True
-        except EC2ResponseError as e:
-            if e.error_code == 'DryRunOperation':
-                module.exit_json(changed=True)
-            module.fail_json(msg="Failed to create route table: {0}".format(e.message),
-                             exception=traceback.format_exc())
+        changed = True
+        if not module.check_mode:
+            try:
+                route_table = connection.create_route_table(VpcId=vpc_id)['RouteTable']
+            except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                module.fail_json_aws(e, msg="Error creating route table")
 
     if routes is not None:
-        try:
-            result = ensure_routes(connection, route_table, routes,
-                                   propagating_vgw_ids, module.check_mode,
-                                   purge_routes)
-            changed = changed or result['changed']
-        except EC2ResponseError as e:
-            module.fail_json(msg="Error while updating routes: {0}".format(e.message),
-                             exception=traceback.format_exc())
+        result = ensure_routes(connection, module, route_table, routes,
+                               propagating_vgw_ids, module.check_mode,
```

Comment:
For things with this many arguments, I'd really like to see keywords used to avoid any order mixups in the future.

Reply:
Done